src/shared/utils.js

```diff
@@ -4,6 +4,29 @@ function convertToSafeString(string, seperator="_"){
 			.toLowerCase();
 }//convertToSafeString
 
+
+function getDeviceWidth(){
+	const w = Math.max(
+		document.documentElement.clientWidth, 
+		window.innerWidth 
+		|| 0
+	);
+
+	return w;
+}// getDeviceWidth
+
+function getDeviceHeight(){
+	const h = Math.max(
+		document.documentElement.clientHeight,
+		window.innerHeight
+		|| 0
+	);
+
+	return h;
+}// getDeviceHeight
```

Comment:
as mentioned above - could be worth combining these into a `getDeviceDimensions` function. Also boooooo single letter variablesssss